File: accudrop/app/src/main/java/me/chrislane/accudrop/db/DateConverter.java
package me.chrislane.accudrop.db;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class DateConverter {
    /**
     * Convert a timestamp into a <code>Date</code> object.
     *
     * @param timestamp The timestamp to convert.
     * @return The matching <code>Date</code> object.
     */
    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    /**
     * Convert a <code>Date</code> object into a timestamp.
     *
     * @param date The date to convert.
     * @return The matching timestamp.
     */
    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/me/chrislane/accudrop/db/converter/DateConverter.kt
package me.chrislane.accudrop.db.converter

import android.arch.persistence.room.TypeConverter
import java.util.*

class DateConverter {
    /**
     * Convert a timestamp into a `Date` object.
     *
     * @param timestamp The timestamp to convert.
     * @return The matching `Date` object.
     */
    @TypeConverter
    fun toDate(timestamp: Long?): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    /**
     * Convert a `Date` object into a timestamp.
     *
     * @param date The date to convert.
     * @return The matching timestamp.
     */
    @TypeConverter
    fun toTimestamp(date: Date?): Long? {
        return date?.time
    }
}
